cardano-cli/src/Cardano/CLI/EraBased/Script/Proposal/Read.hs

```diff
@@ -129,3 +133,25 @@ readProposalScriptWitness (propFp, Just certScriptReq) = do
                       redeemer
                       execUnits
               )
+
+data ProposalError
+  = ProposalErrorFile (FileError CliScriptWitnessError)
+  deriving Show
+
+instance Error ProposalError where
+  prettyError = pshow
+
+readProposal
+  :: Exp.IsEra era
+  => (ProposalFile In, Maybe (ScriptRequirements Exp.ProposalItem))
+  -> IO (Either ProposalError (Proposal era, Maybe (ProposalScriptWitness era)))
```

Comment:
I'm not a fan of `IO (Either e a)`, because it does not compose well. Can we use `CIO` instead? If not, maybe `ExceptT`?